src/main/java/com/minecolonies/coremod/entity/pathfinding/WalkToProxy.java

```diff
@@ -315,7 +315,7 @@ else if (targetY <= levelDepth)
                 if(level.getRandomNode().getParent() != null)
                 {
                     final List<BlockPos> nodesToTarget = new ArrayList<>();
-                    com.minecolonies.coremod.entity.ai.citizen.miner.Node currentNode = level.getNode(level.getRandomNode().getParent());
+                    Node currentNode = level.getNode(level.getRandomNode().getParent());
                     while (new Point2D.Double(currentNode.getX(), currentNode.getZ()) != currentNode.getParent() && currentNode.getParent() != null)
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Use the "equals" method if value comparison was intended. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS1698)